library/database/class.mysqlstructure.php

```diff
@@ -337,9 +340,20 @@ protected function getTableInfo($tableName) {
     }
 
     /**
+     * Given an SQL representing a single basic alter-table query to add indexes, append the default index options.
      *
+     * @param string $sql
+     * @return string
+     */
+    private function indexSqlWithOptions(string $sql): string {
+        $result = preg_replace('/;?(?=\n*$)/', ", " . self::INDEX_OPTIONS . ";", $sql, 1);
```

Comment:
Is there any way to restructure things to not have to do this regex kludge or is that too risky? Like you can modify the index SQL to not end with a semi colon? Otherwise I would do something similar to what Eduardo suggested, but maybe instead: `rtrim(rtrim($sql), ';')`.

Reply:
Modifying the existing SQL is what I was hoping to avoid doing here. If we can avoid that, I would definitely prefer it.

> Otherwise I would do something similar to what Eduardo suggested, but maybe instead: `rtrim(rtrim($sql), ';')`.

No love for my surgical-precision regex? 😩